### PR TITLE
VEN-693 | Customer page: fetching templates for the messages

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description :sparkles:

* Separate CustomerMessageForm to its own feature
* Fetch notification templates and show them on the form
* Add a preview for notification templates

### Closes :no_good_woman:

* [VEN-693](https://helsinkisolutionoffice.atlassian.net/browse/VEN-693): Customer page: fetching templates for the messages

### Automated tests :gear:️

* Improved tests for CustomerMessageForm

### Manual testing :construction_worker_man:

* Open the Customers view
* Select one or many customers
* Click the send notification button on the top table tools
* Test the changed modal

### CODE REVIEW

File: src/features/customerMessageForm/CustomerMessageFormContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  return (
+    <>
+      <CustomerMessageForm
+        templates={notificationTemplates}
+        handleCancel={closeModal}
+        handlePreview={handlePreview}
+        handleSendMessage={onSubmit}
+      />
+      {preview && (
```
Comment (santtul) [inline src/features/customerMessageForm/CustomerMessageFormContainer.tsx:52]: We can remove this wrapping condition. `isOpen={!!preview}` does the trick.

File: src/features/customerMessageForm/CustomerMessageFormContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  selectedCustomerIds: string[];
+}
+
+const CustomerMessageFormContainer = ({
+  closeModal,
+  handleSendMessage,
+  selectedCustomerIds,
+}: CustomerMessageFormContainerProps) => {
+  const { data, loading } = useQuery<NOTIFICATION_TEMPLATES>(NOTIFICATION_TEMPLATES_QUERY);
+  const [preview, setPreview] = useState<string | undefined>(undefined);
```
Comment (santtul) [inline src/features/customerMessageForm/CustomerMessageFormContainer.tsx:25]: Should we call this `previewHtml`?

File: src/features/customerMessageForm/CustomerMessageFormContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  if (loading || !data?.notificationTemplates) return <LoadingSpinner isLoading={true} />;
+
+  const notificationTemplates = getNotificationTemplates(data);
+
+  const handlePreview = (id: string) => {
+    setPreview(
+      notificationTemplates.find((template) => {
```
Comment (santtul) [inline src/features/customerMessageForm/CustomerMessageFormContainer.tsx:33]: Could we extract function here: `getPreviewHtml` ?

File: src/features/customerMessageForm/preview/preview.module.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
+@import 'colours';
+@import 'spacings';
+
+.preview {
+  display: block;
```
Comment (santtul) [inline src/features/customerMessageForm/preview/preview.module.scss:5]: Style has no effect

File: src/features/customerMessageForm/preview/preview.module.scss [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+.iframe {
+  width: 800px;
+  height: 600px;
+  border: 2px solid $black-50;
+}
+
+.buttons {
+  margin-top: units(1);
+  display: block;
```
Comment (santtul) [inline src/features/customerMessageForm/preview/preview.module.scss:16]: Style has no effect

File: src/features/customerMessageForm/CustomerMessageFormContainer.tsx (hunk 1 above)
Comment (TatuArvela) [reply src/features/customerMessageForm/CustomerMessageFormContainer.tsx:52]: Indeed, fixed

File: src/features/customerMessageForm/CustomerMessageFormContainer.tsx (hunk 2 above)
Comment (TatuArvela) [reply src/features/customerMessageForm/CustomerMessageFormContainer.tsx]: Yes, that's a much clearer name. Changed.

File: src/features/customerMessageForm/CustomerMessageFormContainer.tsx (hunk 3 above)
Comment (TatuArvela) [reply src/features/customerMessageForm/CustomerMessageFormContainer.tsx]: Sure! Extracted.

File: src/features/customerMessageForm/preview/preview.module.scss (hunk 4 above)
Comment (TatuArvela) [reply src/features/customerMessageForm/preview/preview.module.scss]: 💅 🚮 👍

File: src/features/customerMessageForm/preview/preview.module.scss (hunk 5 above)
Comment (TatuArvela) [reply src/features/customerMessageForm/preview/preview.module.scss]: 💅 🚮 👍

**Pull request merged**